### PR TITLE
Implement composition (as an example of the design pattern).

### PR DESCRIPTION
**Pull request by marilirulita**
### Implementing the Decorator design pattern to use two decorators in order to capitalize and trim people's names.

### CODE REVIEW

File: decorate.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
```
Comment (mikethreels) [inline decorate.rb]: - [x] this method should also provide a message, which will look something like below:
```
def correct_name
        raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
    end
```

File: decorate.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
```
Comment (mikethreels) [inline decorate.rb]: - [x] in the Decorator class you should also include the attribute accessor for the `@nameable`.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Mar, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: decorate.rb (hunk 2 above)
Comment (marilirulita) [reply decorate.rb]: Thanks for the quick answer @mikethreels 
Got it, I will add these

**Review by KDlamini — APPROVED** [review]
### Hi @marilirulita 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

## HIGHLIGHTS 🌟 

✔️ Required methods and classes are created
✔️ Inheritance is implemented
✔️ No linter errors

# Status: Approved ✔️ 
 
simon

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**